websites/A/AnimeDekho/presence.ts

```diff
@@ -1,79 +1,254 @@
+        presenceData.smallImageText = 'Paused'
+        delete presenceData.startTimestamp
+        delete presenceData.endTimestamp
+      }
+    }
+
+    if (buttons) {
+      presenceData.buttons = [
+        {
+          label: 'Watch Movie',
+          url: href,
+        },
+      ]
+    }
+  }
+
+  // Watch page
+  else if (pathname.includes('/epi/')) {
+    const rawtitle = document.querySelector('.entry-title')?.textContent || ''
+    const title = rawtitle.replace(/Season\s\d+\sEpisode\s\d+|\d+x\d+$/i, '').trim()
+    const episode = document.querySelector('.season-episode')?.textContent || ''
+    const thumbnail = document.querySelector('.post-thumbnail img')?.getAttribute('src')
+
+    presenceData.details = title || 'Unknown Series'
+    presenceData.state = episode || 'Unknown Episode'
+
+    if (thumbnail) {
+      presenceData.largeImageKey = String(thumbnail)
+      presenceData.largeImageText = title || 'Unknown Series'
+    }
+
+    // Handle video data for timestamps
+    if (data && timestamps) {
+      if (!data.paused && data.currTime > 0 && data.duration > 0) {
+        // Video is playing - calculate timestamps only when needed
+        const [startTimestamp, endTimestamp] = getTimestamps(
+          Math.floor(data.currTime),
+          Math.floor(data.duration),
+        )
+        presenceData.startTimestamp = startTimestamp
+        presenceData.endTimestamp = endTimestamp
+        presenceData.smallImageKey = SmallImageAssets.Play
+        presenceData.smallImageText = 'Watching'
+      }
+      else {
+        // Video is paused
+        presenceData.smallImageKey = SmallImageAssets.Pause
+        presenceData.smallImageText = 'Paused'
+        delete presenceData.startTimestamp
+        delete presenceData.endTimestamp
+      }
+    }
+
+    if (buttons) {
+      // Simplified button creation with safer path handling
+      presenceData.buttons = [{ label: 'Watch Episode', url: href }]
+
+      // Try to extract anime ID for "View Series" button
+      try {
+        const match = pathname.match(/\/epi\/([^/]+)/)
+        if (match && match[1]) {
+          const animeId = match[1].replace(/-\d+x\d+$/, '')
+          presenceData.buttons.push({
+            label: 'View Series',
+            url: `${document.location.origin}/series/${animeId}`,
+          })
+        }
+      }
+      catch {
+        // If anything fails, we already have the "Watch Episode" button
+      }
+    }
   }
-  else {
-    presenceData.details = 'Browsing AnimeDekho'
+
+  // Add this check right here, before the final setActivity calls
+  if (!buttons && presenceData.buttons) {
+    delete presenceData.buttons
   }
 
-  presenceData.largeImageKey = ActivityAssets.DefaultLogo
-  presence.setActivity(presenceData)
+  // These should be the last lines of your UpdateData event handler
+  if (presenceData.details)
+    presence.setActivity(presenceData)
+
+  else
+    presence.setActivity()
```

Comment:
Other than this, LGTM
```suggestion
    presence.clearActivity()
```